### ISSUE TITLE
ActionView::Template::Error undefined method `any?' for nil:NilClass in panel_component.html.erb

### ISSUE THREAD

**Issue by johlym** [avo-hq/avo#2047]
### Describe the bug
Avo 3 doesn't support sidebars for the community/free version, so a Resource should never have any `sidebars`. With `panel_component.html.erb` using `sidebars.any?` ([ref](https://github.com/avo-hq/avo/blame/main/app/components/avo/items/panel_component.html.erb#L14)) as the conditional for whether to display sidebars, it doesn't take into account `sidebars` being `nil` (intended or otherwise), and as such, `.any?` returns `undefined method 'any?' for nil:NilClass`.

Seems like one of two things needs to happen here to solve this. Either `sidebars` equals _something_ regardless of the Pro license status, when no [sidebar fields](https://docs.avohq.io/3.0/resource-sidebar.html#adding-fields-to-the-sidebar) are defined in the resource–be it `[]` or `""`–or the component should be updated to `sidebars&.any?` to support `sidebars`' `nil` state.

### Steps to Reproduce

My reproducing steps are convoluted because I'm running into this after a v2 to v3 upgrade. I don't get this error locally where Pro features are enabled and usable, only in production where Avo runs in community/free mode. However, I receive this error on 100% of the existing and new resources I create. The error only manifests when the app is run in a `production` environment.

### Expected behavior & Actual behavior
Since I do not define sidebar fields for a resource, I should only expect to see the regularly defined `fields` and no error regarding the resource sidebar.

### Models and resource files

Here's one of my resources that triggers this. Note that all of my resources trigger this and none of them are have `def sidebars; ... ; end`

```rb
class Avo::Resources::Note < Avo::BaseResource
  self.title = :code

  self.after_create_path = :index
  self.after_update_path = :index
  # self.search_query = -> do
  #   query.ransack(id_eq: params[:q], m: "or").result(distinct: false)
  # end

  def fields
    field :id, as: :id
    field :code, as: :text
    field :detail, as: :text
  end
end
```

### System configuration
**Avo version**: 3.0.2

**Rails version**: 7.0.4.3

**Ruby version**: 3.2.2

**License type**:

 - [x] Community
 - [ ] Pro
 - [ ] Advanced

**Are you using Avo monkey patches, overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [x] No

#### Impact

 - [ ] High impact (It makes my app un-usable.)
 - [x] Medium impact (I'm annoyed, but I'll live.) (This prevents an upgrade from 2.x to 3.x)
 - [ ] Low impact (It's really a tiny thing that I could live with.)

#### Urgency

 - [ ] High urgency (I can't continue development without it.)
 - [x] Medium urgency (I found a workaround, but I'd love to have it fixed.)
 - [ ] Low urgency (It can wait. I just wanted you to know about it.)

### Stack Trace

```
NoMethodError: undefined method `any?' for nil:NilClass (NoMethodError)
  from avo (3.0.2) app/components/avo/items/panel_component.html.erb:14:in `block in call'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `capture'
  from view_component (3.7.0) lib/view_component/base.rb:274:in `content'
  from view_component (3.7.0) lib/view_component/slotable.rb:328:in `get_slot'
  from view_component (3.7.0) lib/view_component/slotable.rb:92:in `block in renders_one'
  from avo (3.0.2) app/components/avo/panel_component.html.erb:19:in `block in call'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `capture'
  from actionview (7.0.4.3) lib/action_view/helpers/tag_helper.rb:371:in `content_tag'
  from avo (3.0.2) app/components/avo/panel_component.html.erb:1:in `call'
  from view_component (3.7.0) lib/view_component/compiler.rb:120:in `render_template_for'
  from view_component (3.7.0) lib/view_component/base.rb:106:in `render_in'
  from actionview (7.0.4.3) lib/action_view/helpers/rendering_helper.rb:42:in `render'
  from view_component (3.7.0) lib/view_component/base.rb:189:in `render'
  from avo (3.0.2) app/components/avo/items/panel_component.html.erb:1:in `call'
  from view_component (3.7.0) lib/view_component/compiler.rb:120:in `render_template_for'
  from view_component (3.7.0) lib/view_component/base.rb:106:in `render_in'
  from actionview (7.0.4.3) lib/action_view/helpers/rendering_helper.rb:42:in `render'
  from view_component (3.7.0) lib/view_component/base.rb:189:in `render'
  from avo (3.0.2) app/components/avo/items/switcher_component.html.erb:2:in `call'
  from view_component (3.7.0) lib/view_component/compiler.rb:120:in `render_template_for'
  from view_component (3.7.0) lib/view_component/base.rb:106:in `render_in'
  from actionview (7.0.4.3) lib/action_view/helpers/rendering_helper.rb:42:in `render'
  from view_component (3.7.0) lib/view_component/base.rb:189:in `render'
  from avo (3.0.2) app/components/avo/views/resource_edit_component.html.erb:24:in `block (4 levels) in call'
  from avo (3.0.2) app/components/avo/views/resource_edit_component.html.erb:23:in `each'
  from avo (3.0.2) app/components/avo/views/resource_edit_component.html.erb:23:in `each_with_index'
  from avo (3.0.2) app/components/avo/views/resource_edit_component.html.erb:23:in `block (3 levels) in call'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `capture'
  from actionview (7.0.4.3) lib/action_view/helpers/tag_helper.rb:371:in `content_tag'
  from avo (3.0.2) app/components/avo/views/resource_edit_component.html.erb:22:in `block (2 levels) in call'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `capture'
  from actionview (7.0.4.3) lib/action_view/helpers/form_helper.rb:768:in `form_with'
  from avo (3.0.2) app/components/avo/views/resource_edit_component.html.erb:12:in `block in call'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
  from actionview (7.0.4.3) lib/action_view/helpers/capture_helper.rb:45:in `capture'
  from actionview (7.0.4.3) lib/action_view/helpers/tag_helper.rb:371:in `content_tag'
  from avo (3.0.2) app/components/avo/views/resource_edit_component.html.erb:1:in `call'
  from view_component (3.7.0) lib/view_component/compiler.rb:120:in `render_template_for'
  from view_component (3.7.0) lib/view_component/base.rb:106:in `render_in'
  from actionview (7.0.4.3) lib/action_view/helpers/rendering_helper.rb:42:in `render'
  from avo (3.0.2) app/views/avo/base/new.html.erb:1
...[truncated]...
```

**Comment by adrianthedev** [comment]
Thanks @johlym. We'll have a look tomorrow, fix it and put out a patch release.

**Comment by xeron** [comment]
I'm getting the same error. Annoying part is this issue wasn't caught during testing because it worked fine locally and in staging.

**Comment by adrianthedev** [comment]
Got it! @Paul-Bob is working on this today and releasing a fix. Thanks for reporting it!

**Comment by Paul-Bob** [comment]
Thanks for the detailed report @johlym and for the confirmation @xeron!

It's fixed on the fresh [`3.0.4`](https://rubygems.org/gems/avo/versions/3.0.4) release

**Comment by xeron** [comment]
> It's fixed on the fresh [`3.0.4`](https://rubygems.org/gems/avo/versions/3.0.4) release

Not sure what's up but my bundler doesn't see 3.0.4 release.

**Comment by johlym** [comment]
Rubygems caching can take a bit to refresh.On Nov 23, 2023, at 11:13 AM, Ivan Larionov ***@***.***> wrote:﻿

It's fixed on the fresh 3.0.4 release

Not sure what's up but my bundler doesn't see 3.0.4 release.

**Comment by xeron** [comment]
> Rubygems caching can take a bit to refresh

It's been 14 hours since the release and my bundler still doesn't see it. I don't think it's rubygems cache…

**Comment by xeron** [comment]
This is what bundler uses and 3.0.4 is not listed there: https://index.rubygems.org/info/avo

CC @Paul-Bob

**Comment by johlym** [comment]
I'm 100% confident this is on the RubyGems team, but it is something the Avo team can likely bring up with them. I've seen this happen before with other gem releases. It is definitely [published](https://rubygems.org/gems/avo) and also appears in [the full index](https://index.rubygems.org/versions).


Running

```sh
$ bundle install --full-index
```

might be necessary here to pull the complex index down from RubyGems instead. It's slow, but it has always worked for me in the past when there was an inconsistency.

![CleanShot 2023-11-23 at 21 09 12](https://github.com/avo-hq/avo/assets/5686227/4816c241-d9c7-461d-812e-8c9df6988a20)

**Comment by xeron** [comment]
> ```shell
> --full-index
> ```

Thanks, this is helpful locally but things like CI/CD still fail because many scripts just call `bundle install`.

**Comment by Paul-Bob** [comment]
> not is not

Can you provide the error log please?

**Comment by xeron** [comment]
> @xeron just released `3.0.5` hopping that it will solve the issue, also opened an issue on rubygems
> 
> Let me know if `3.0.5` is fetched as expected

Yep all good now. Thanks!
